src/main/java/io/nats/client/impl/NatsConnection.java

```diff
@@ -748,7 +793,7 @@ public Message request(String subject, byte[] body, Duration timeout) throws Int
         Future<Message> incoming = this.request(subject, body);
         try {
             reply = incoming.get(timeout.toNanos(), TimeUnit.NANOSECONDS);
-        } catch (ExecutionException|TimeoutException e) {
+        } catch (ExecutionException | TimeoutException e) {

```

Comment:
Did you intend to leave interrupted and cancellation exceptions to be uncaught?

Reply:
yes, interrupted exceptions should rarely be eaten as they tell the caller the thread was interrupted. Cancellation has the same meaning here for the future, so we don't want to eat it.